### PR TITLE
bug; infinite loop in server initialization if server owner is offline

### PR DESCRIPTION
**Pull request by stoodfarback**

### CODE REVIEW

**Comment by stoodfarback** [comment]
When connecting to discord and initializing the `Server` objects, if the owner of a server is not listed with the members (this happens if the server has more than 100 members and the owner is offline), the code gets stuck in a loop of initializing `Server` -> fetching server owner info -> initializing `Server`.

This fixes it by passing the `Server` object along to the member lookup, so it can use that instead of trying to find it in the cache.

Likely fixes #169 and #170.

**Comment by meew0** [comment]
Thanks for debugging this; it's one of the more pressing issues with 2.1.3 and I had no idea it had to do with the server owner. I'll make a couple comments to code style but functionality wise this seems good.

File: lib/discordrb/cache.rb [hunk 1]
```diff
@@ -85,15 +85,16 @@ def server(id)
       @servers[id] = server
     end
 
-    # Gets a member by both IDs
-    # @param server_id [Integer] The server ID for which a member should be resolved
+    # Gets a member by both IDs, or `Server` and user ID.
+    # @param server_or_id [Integer] The `Server` or server ID for which a member should be resolved
```
Comment (greenbigfrog) [inline lib/discordrb/cache.rb]: This doesn't expect an `[Intger]` anymore

Comment (meew0) [inline lib/discordrb/cache.rb]: Here, the type parameter should now be `[Integer, Server]`.

Comment (stoodfarback) [inline lib/discordrb/cache.rb]: Ah, missed that. Updated pr.

File: lib/discordrb/cache.rb [hunk 2]
```diff
@@ -85,15 +85,16 @@ def server(id)
       @servers[id] = server
     end
 
-    # Gets a member by both IDs
-    # @param server_id [Integer] The server ID for which a member should be resolved
+    # Gets a member by both IDs, or `Server` and user ID.
+    # @param server_or_id [Server, Integer] The `Server` or server ID for which a member should be resolved
     # @param user_id [Integer] The ID of the user that should be resolved
     # @return [Member, nil] The member identified by the IDs, or `nil` if none could be found
```
Comment (greenbigfrog) [inline lib/discordrb/cache.rb]: could you please decapitalize the return if you're at it? looks weird in the cods if it's not... (`The` -> `the`)

Comment (stoodfarback) [inline lib/discordrb/cache.rb]: Sorry, I don't understand what you mean.

**Pull request merged**